src/core/Akka/Dispatch/ActorTaskScheduler.cs

```diff
@@ -21,11 +21,26 @@ namespace Akka.Dispatch
     public class ActorTaskScheduler : TaskScheduler
     {
         private readonly ActorCell _actorCell;
+
+        private Action _beforeTaskStarted;
+        private Action _afterTaskCompleted;
+        
         /// <summary>
         /// TBD
         /// </summary>
         public object CurrentMessage { get; private set; }
 
+        /// <summary>
+        /// Wraps async task scheduling with pre/post actions
+        /// </summary>
+        /// <param name="beforeStartStarted">Executed before scheduling the task call/></param>
+        /// <param name="afterTaskCompleted">Executed in task continuation after task completion</param>
+        internal void WrapTaskScheduling(Action beforeStartStarted = null, Action afterTaskCompleted = null)
```

Comment:
Better to have the `TestActorRef` subclass the `ActorTaskScheduler` with a custom implementation that does this rather than introduce mutable state inside the `ActorTaskScheduler` implementation shared / used by all actors.